tests/Drupal/Commenting/FunctionCommentUnitTest.inc

```diff
@@ -544,3 +544,293 @@ class Test40 {
     }
 
 }
+
+/**
+ * Generics test.
+ *
+ * @param T $object
+ *   Template.
+ *
+ * @template T
+ *
+ * @return T
+ */
+function test40(object $object) {
+  return clone $object;
+}
+
+/**
+ * PHPStan: Basic types.
+ *
+ * @param int $param1
+ *   Parameter.
+ * @param string $param2
+ *   Parameter.
+ * @param array-key $param3
+ *   Parameter.
+ * @param bool $param4
+ *   Parameter.
+ * @param true $param5
+ *   Parameter.
+ * @param false $param6
+ *   Parameter.
+ * @param null $param7
+ *   Parameter.
+ * @param float $param8
+ *   Parameter.
+ * @param double $param9
+ *   Parameter.
+ * @param scalar $param10
+ *   Parameter.
+ * @param array $param11
+ *   Parameter.
+ * @param iterable $param12
+ *   Parameter.
+ * @param callable $param13
+ *   Parameter.
+ * @param resource $param14
+ *   Parameter.
+ * @param void $param15
+ *   Parameter.
+ * @param object $param16
+ *   Parameter.
+ *
+ * @see https://phpstan.org/writing-php-code/phpdoc-types#basic-types
+ */
+function test41(int $param1, string $param2, mixed $param3, bool $param4, mixed $param5, mixed $param6, mixed $param7, float $param8, double $param9, mixed $param10, array $param11, iterable $param12, callable $param13, mixed $param14, mixed $param15, object $param16) {
```

Comment:
$param5 type should be bool? We should not give bad examples with using mixed when this should be bool. Same for others, especially arrays.

Reply:
No its correctly and its explicitly a value type phpstan and psalm support. Like false it doesn't really make sense the way its used in this example but its provided for completeness so you can use it in a union type.

I've seen a bit of discussion about allowing true as a type along with false natively but I don't think its made it to an RFC since its admittedly a pretty odd case so I don't think anyone has the itch to scratch so static analysis hits like this are the only way to do it currently.